src/components/apphost.js

```diff
@@ -426,33 +426,26 @@ define(["appSettings", "browser", "events", "htmlMediaHelper", "webSettings"], f
         }
     };
 
-    var doc = self.document;
     var isHidden = false;
 
-    if (doc) {
-        if (void 0 !== doc.visibilityState) {
+    if (self.document) {
+        if (self.document.visibilityState !== undefined) {
             visibilityChange = "visibilitychange";
             visibilityState = "hidden";
-        } else {
-            if (void 0 !== doc.mozHidden) {
-                visibilityChange = "mozvisibilitychange";
-                visibilityState = "mozVisibilityState";
-            } else {
-                if (void 0 !== doc.msHidden) {
-                    visibilityChange = "msvisibilitychange";
-                    visibilityState = "msVisibilityState";
-                } else {
-                    if (void 0 !== doc.webkitHidden) {
-                        visibilityChange = "webkitvisibilitychange";
-                        visibilityState = "webkitVisibilityState";
-                    }
-                }
-            }
+        } else if (self.document.mozHidden !== undefined) {
+            visibilityChange = "mozvisibilitychange";
+            visibilityState = "mozVisibilityState";
+        } else if (self.document.msHidden !== undefined) {
+            visibilityChange = "msvisibilitychange";
+            visibilityState = "msVisibilityState";
+        } else if (self.document.webkitHidden !== undefined) {
+            visibilityChange = "webkitvisibilitychange";
+            visibilityState = "webkitVisibilityState";
         }
     }
 
-    if (doc) {
-        doc.addEventListener(visibilityChange, function () {
+    if (self.document) {
+        document.addEventListener(visibilityChange, function () {
             if (document[visibilityState]) {
```

Comment:
ESLint considers `document.visibilityState` to be requested. :confused: 
```suggestion
            /* eslint-disable-next-line compat/compat */
            if (document[visibilityState]) {
```